### ISSUE TITLE
unused_import_linter() shouldn't throw for the tested package itself

### ISSUE THREAD

**Issue by MichaelChirico** [r-lib/lintr#1978]
```r
lint("tests/testthat.R")
# tests/testthat.R:2:1: warning: [unused_import_linter] Package 'lintr' is attached but never used.
# library(lintr)
# ^~~~~~~~~~~~~~
```

This is of course standard practice for testthat.R files, and is part of the {usethis}-endorsed template:

https://github.com/r-lib/usethis/blob/d3614c701b9e2ee6f680a245d3315b1b941f49de/inst/templates/testthat.R#L10

**Comment by jameslamb** [comment]
I ran into this today with a `testthat.R`.

Just wanted to add here that calling `library(pkg-being-tested)` is also recommended in the "testing" section of the R Packages book: https://r-pkgs.org/testing-basics.html#initial-setup

<img width="839" alt="image" src="https://github.com/r-lib/lintr/assets/7608904/d81eb13c-058e-4e81-b14e-1465692d80bc">

That's another reason it's so common, I think.
